multi-cluster-orchestrator/samples/hello_world/1-infrastructure/main.tf

```diff
@@ -0,0 +1,298 @@
+
+  fleet {
+    project = data.google_project.default.project_id
+  }
+
+  gateway_api_config {
+    channel = "CHANNEL_STANDARD"
+  }
+
+  workload_identity_config {
+    workload_pool = "${data.google_project.default.project_id}.svc.id.goog"
+  }
+
+  cluster_autoscaling {
+    auto_provisioning_defaults {
+      service_account = google_service_account.clusters["hub"].email
+    }
+  }
+
+  # Set `deletion_protection` to `true` will ensure that one cannot
+  # accidentally delete this instance by use of Terraform.
+  deletion_protection = false
+}
+
+### Multi-Cluster Gateway
+resource "google_gke_hub_feature" "multiclusteringress" {
+  name     = "multiclusteringress"
+  location = "global"
+  spec {
+    multiclusteringress {
+      config_membership = trimprefix(google_container_cluster.hub.fleet[0].membership, "//gkehub.googleapis.com/")
+    }
+  }
+}
+
+### Logging Metric
+resource "google_logging_metric" "failed_scale_up_metric" {
+  name   = "ClusterAutoscalerFailedScaleUp"
+  filter = "jsonPayload.reason=FailedScaleUp AND jsonPayload.reportingComponent=cluster-autoscaler"
+  label_extractors = {
+    cluster        = "EXTRACT(resource.labels.cluster_name)"
+    location       = "EXTRACT(resource.labels.location)"
+    namespace_name = "EXTRACT(resource.labels.namespace_name)"
+    pod_name       = "EXTRACT(resource.labels.pod_name)"
+    project_id     = "EXTRACT(resource.labels.project_id)"
+  }
+  metric_descriptor {
+    metric_kind  = "DELTA"
+    display_name = "metric reflecting occurences of FailedScaleUp"
+    value_type   = "INT64"
+    labels {
+      key = "namespace_name"
+    }
+    labels {
+      key = "cluster"
+    }
+    labels {
+      key = "pod_name"
+    }
+    labels {
+      key = "project_id"
+    }
+    labels {
+      key = "location"
+    }
+  }
+}
+
+### Helm Releases
+data "google_client_config" "default" {}
+
+provider "helm" {
+  kubernetes {
+    host                   = "https://${google_container_cluster.hub.endpoint}"
+    token                  = data.google_client_config.default.access_token
+    cluster_ca_certificate = base64decode(google_container_cluster.hub.master_auth[0].cluster_ca_certificate)
+  }
+}
+
+resource "helm_release" "argocd" {
+  name             = "argocd"
+  chart            = "https://github.com/argoproj/argo-helm/releases/download/argo-cd-${local.argocd_version}/argo-cd-${local.argocd_version}.tgz"
+  namespace        = "argocd"
+  create_namespace = true
+  timeout          = 1200
+}
+
+# Provide guidance on accessing ArgoCD UI
+output "argocd" {
+  description = "ArgoCD Server UI Access"
+  value       = helm_release.argocd.metadata[0].notes
+}
+
+# NOTE: Helm does not support upgrading CRDs at this time.
+# https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations
+resource "helm_release" "orchestrator" {
+  name       = "orchestrator"
+  repository = "https://googlecloudplatform.github.io/gke-fleet-management"
```

Comment:
Don't we need `/fleet-charts` in here? Ditto for the other helm_releases

Reply:
This repo is also a Helm Chart Repository, so it uses this index to locate the chart asset (based on name and version, rather than path) to download: https://googlecloudplatform.github.io/gke-fleet-management/index.yaml